docs/specification/draft/basic/authorization.mdx

```diff
@@ -501,6 +618,66 @@ Authorization servers **MUST** take precautions to prevent redirecting user agen
 
 Authorization servers **SHOULD** only automatically redirect the user agent if it trusts the redirection URI. If the URI is not trusted, the authorization server MAY inform the user and rely on the user to make the correct decision.
 
+### Client ID Metadata Document Security
+
+When implementing Client ID Metadata Documents, authorization servers **MUST** consider the security implications
+detailed in [OAuth Client ID Metadata Document, Section 6](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-client-id-metadata-document-00#name-security-considerations).
+Key considerations include:
+
+#### SSRF Protection
+
+Authorization servers fetching metadata documents **MUST** protect against Server-Side Request Forgery (SSRF) attacks:
+
+- Validate URLs and resolved IP addresses before fetching
+- Limit response size (recommended 5 kilobytes)
+- Implement request timeouts
+- Rate limit outbound metadata fetches
+- Never cache error responses or invalid documents
+
+The authorization server takes a URL as input from an unknown client and fetches that URL.
+A malicious client could use this to send non-metadata requests on behalf of the authorization server,
+such as requests to private administration endpoints the authorization server has access to.
+
+#### DDoS Protection
+
+Authorization servers **MUST** protect against being used as DDoS amplifiers:
+
+- Implement aggressive caching of metadata documents (respecting HTTP cache headers)
+- Rate limit metadata fetch requests per client
+- Monitor and alert on unusual metadata fetch patterns
+- Only fetch client metadata after authenticating the user
+
+While there is no amplification in the fetch request bandwidth, aggressive caching minimizes
```

Comment:
Since there is no DDoS amplification occurring, I think it would make more sense to call this whole section `DoS` instead of `DDoS`. Then you can move things like request timeouts and max response size from above into this section as well.